resources/node.rb

```diff
@@ -76,20 +78,30 @@ def to_array(var)
   # Install Failover Clustering feature and PowerShell module
   windows_features = %w(Failover-Clustering RSAT-Clustering-Powershell)
   windows_features << 'RSAT-Clustering-Mgmt' if new_resource.install_tools
-  install_windows_feature(windows_features)
+  install_windows_features(windows_features)
+
   # Create the cluster
   powershell_out_with_options!("New-Cluster -Name #{new_resource.cluster_name} -Node #{node['hostname']} -StaticAddress #{new_resource.cluster_ip} -Force") unless cluster_exist?(new_resource.cluster_name)
   # Add any available disks to the cluster
   powershell_out_with_options('Get-ClusterAvailableDisk | Add-ClusterDisk')
-  # Set quorum to use node & disk majority
-  powershell_out_with_options!("Set-ClusterQuorum -NodeAndDiskMajority \'#{new_resource.quorum_disk}\'") unless cluster_quorum_disk?(new_resource.quorum_disk)
+
+  if new_resource.quorum_disk && new_resource.fs_witness
+    Chef::Log.fatal('You provided both quorum_disk and fs_witness, only one is supported!')
+  elsif new_resource.quorum_disk
+    # Set quorum to use node & disk majority
+    powershell_out_with_options!("Set-ClusterQuorum -NodeAndDiskMajority \'#{new_resource.quorum_disk}\'") unless cluster_quorum_disk?(new_resource.quorum_disk)
+  elsif new_resource.fs_witness
+    # Set witness using FileShare
+    powershell_out_with_options!("Set-ClusterQuorum -NodeAndFileShareMajority \'#{new_resource.fs_witness}\'") unless cluster_quorum_fs_witness?
+  end
```

Comment:
This can be broken up:

```
  if new_resource.quorum_disk && new_resource.fs_witness
    Chef::Log.fatal('You provided both quorum_disk and fs_witness, only one is supported!')
  end

  if new_resource.quorum_disk
    # Set quorum to use node & disk majority
    powershell_out_with_options!("Set-ClusterQuorum -NodeAndDiskMajority \'#{new_resource.quorum_disk}\'") unless cluster_quorum_disk?(new_resource.quorum_disk)
  elsif new_resource.fs_witness
    # Set witness using FileShare
    powershell_out_with_options!("Set-ClusterQuorum -NodeAndFileShareMajority \'#{new_resource.fs_witness}\'") unless cluster_quorum_fs_witness?
  end
```

Makes me wonder though if we shouldn't change the params, and have a `quorum_type` (e.g. `fileshare` or `disk`) and a `quorum_name` or `quorum_path` param instead?

Then we could do something like:

```
# Return if correct quorum is already set
return if cluster_quorum_disk?(new_resource.quorum_path) || cluster_quorum_fileshare?(new_resource.quorum_path)

# Set correct quorum if not
set_quorum_cmd = "Set-ClusterQuorum"
set_quorum_cmd << "  -NodeAndDiskMajority \'#{new_resource.quorum_path}\'" if new_resource.quorum_type == 'disk'
set_quorum_cmd << "  -NodeAndFileShareMajority \'#{new_resource.quorum_path}\'" if new_resource.quorum_type == 'fileshare'
powershell_out_with_options!(set_quorum_cmd)
```

We could then specify acceptable param values.

Thoughts?

Reply:
Looks nicer indeed. However could we move this for a next version?